packages/cactus-plugin-htlc-eth-besu-erc20/src/main/typescript/plugin-htlc-eth-besu-erc20.ts

```diff
@@ -0,0 +1,320 @@
+import { Optional } from "typescript-optional";
+
+import {
+  IPluginWebService,
+  ICactusPlugin,
+  ICactusPluginOptions,
+  IWebServiceEndpoint,
+  PluginAspect,
+} from "@hyperledger/cactus-core-api";
+import { Checks, LogLevelDesc } from "@hyperledger/cactus-common";
+import {
+  EthContractInvocationType,
+  InvokeContractV1Response,
+  PluginLedgerConnectorBesu,
+  RunTransactionResponse,
+  Web3SigningCredentialType,
+} from "@hyperledger/cactus-plugin-ledger-connector-besu";
+import { GetSingleStatusEndpoint } from "./web-services/get-single-status-endpoint";
+import { GetStatusEndpoint } from "./web-services/get-status-endpoint";
+import { NewContractEndpoint } from "./web-services/new-contract-endpoint";
+import { RefundEndpoint } from "./web-services/refund-endpoint";
+import { WithdrawEndpoint } from "./web-services/withdraw-endpoint";
+import { InitializeEndpoint } from "./web-services/initialize-endpoint";
+import HashedTimeLockContractJSON from "../solidity/contracts/HashedTimeLockContract.json";
+import { PluginRegistry } from "@hyperledger/cactus-core";
+import { NewContractRequest } from ".";
+import HashTimeLockJSON from "../solidity/contracts/HashedTimeLockContract.json";
+import {
+  RefundRequest,
+  WithdrawRequest,
+} from "./generated/openapi/typescript-axios";
+
+export interface IPluginHtlcEthBesuErc20Options extends ICactusPluginOptions {
+  instanceId: string;
+  logLevel?: LogLevelDesc;
+  pluginRegistry: PluginRegistry;
+}
+export class PluginHtlcEthBesuErc20
+  implements ICactusPlugin, IPluginWebService {
+  public static readonly CLASS_NAME = "PluginHtlcEthBesuErc20";
+  private readonly instanceId: string;
+  private readonly pluginRegistry: PluginRegistry;
+  private readonly estimatedGas = 6721975;
+  private readonly connectorPackageName =
+    "@hyperledger/cactus-plugin-ledger-connector-besu";
+
+  constructor(public readonly opts: IPluginHtlcEthBesuErc20Options) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(opts, `${fnTag} opts`);
+    Checks.truthy(opts.instanceId, `${fnTag} opts.instanceId`);
+    Checks.truthy(opts.pluginRegistry, `${fnTag} opts.pluginRegistry`);
+    Checks.nonBlankString(opts.instanceId, `${fnTag} opts.instanceId`);
+    this.instanceId = opts.instanceId;
+    this.pluginRegistry = opts.pluginRegistry;
+  }
+
+  public get className(): string {
+    return PluginHtlcEthBesuErc20.CLASS_NAME;
+  }
+
+  /**
+   * Feature is deprecated, we won't need this method in the future.
+   */
+  public getHttpServer(): Optional<Server | SecureServer> {
+    return Optional.empty();
+  }
+
+  /**
+   * Feature is deprecated, we won't need this method in the future.
+   */
+  public async shutdown(): Promise<void> {
+    return;
+  }
+
+  public getInstanceId(): string {
+    return this.instanceId;
+  }
+
+  public getPackageName(): string {
+    return "@hyperledger/cactus-plugin-htlc-eth-besu-erc20";
+  }
+
+  public getAspect(): PluginAspect {
```

Comment:
@jordigiam For now, in order to unblock this from getting the tests executed by the CI, I recommend this little hack to appease the compiler (which will allow the build to succeed, tests to run, etc.).

They are the same at runtime because Typescript enums are just syntactic sugar that ends up being a simple string once the code is transpiled down to Javascript. Please also leave a // FIXME comment for this method to make the change once the other PR is merged and we can put it back the way it is now referencing the actual enum instead of just a plain string.

  public getAspect(): PluginAspect {
    return "HTLC" as PluginAspect;
  }

Reply:
Done